### PR TITLE
Add module to read local system unique identifier

### PR DESCRIPTION
**Pull request by jlongever**
@RackHD/corecommitters @uppalk1 @tannoa2 @keedya @zyoung51 @geoff-reid 

This PR is required for upcoming Redfish ServiceRoot fix and SSDP/UPnP PR.

### CODE REVIEW

File: lib/common/system-uuid.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+        // Attempt to get the system's UUID from sysfs otherwise generate one
+        return fs.readFileAsync('/sys/class/dmi/id/product_uuid')
+        .then(function(content) {
+            if(content) {
+                return content.toString().toLowerCase().replace(/(\r\n|\n|\r)/gm,'');
+            }
+            logger.warning('Error getting system UUID, generating...');
+            return uuid('v4');             
+        })
+        .then(function(uuid) {
```
Comment (zyoung51) [inline lib/common/system-uuid.js]: Do you need a .catch() to handle file read errors like EACCES?

File: spec/lib/common/system-uuid-spec.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
+// Copyright 2016, EMC, Inc.
+'use strict';
+
+describe("SystemUuid", function() {
+    var subject;
+    var regEx = /^[a-fA-F0-9]{8}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{12}$/;
```
Comment (yyscamper) [inline spec/lib/common/system-uuid-spec.js]: This regex is not the same with https://github.com/RackHD/on-core/blob/master/lib/common/constants.js#L197

**Comment by yyscamper** [comment]
:+1: after change the regex to the common one.

File: spec/lib/common/system-uuid-spec.js (hunk 2 above)
Comment (jlongever) [inline spec/lib/common/system-uuid-spec.js]: It is the same as assert-plus's assert.uuid().   https://github.com/mcavage/node-assert-plus/blob/master/assert.js#L12

Are you suggesting we're not asserting UUID values correctly and shouldn't be using assert.uuid()?

**Comment by jlongever** [comment]
test this please

File: spec/lib/common/system-uuid-spec.js (hunk 2 above)
Comment (brianparry) [inline spec/lib/common/system-uuid-spec.js]: @jlongever, @dalebremner added a regex for UUID to Constants.Regex.  I think Felix is suggesting you use that one.  At first glace, they appear the same.

**Pull request merged**